src/uniswap-summaries.md

```diff
@@ -3155,6 +3155,154 @@ module SOLIDITY-UNISWAP-SWAP-SUMMARY
+       <summarize> true </summarize>
+       <env> ( _ (amount1Out |-> var ( Ia1 , uint256 ))
+                 (balance1 |-> var ( Ib1 , uint256 ))
+                 (reserves |-> var ( Ir , uint112 [ ] )) ) #as ENV =>
+             ENV [ amount1In <- var(size(STORE), uint256)]
+       </env>
+       <store> ( _ [ Ia1 <- Va1:MInt{256} ]
+                   [ Ib1 <- Vb1:MInt{256} ]
+                   [ Ir <- ListItem(_) ListItem(Vr1:MInt{112}) ListItem(_) ]
+               ) #as STORE => STORE ListItem(Vb1 -MInt (roundMInt(Vr1)::MInt{256} -MInt Va1))
+       </store>
+       <this-type> uniswapV2Pair </this-type>
+       <current-function> swap </current-function> [priority(40)]
+
+  // Requirements after ternary operators and args to fidUpdate
+  rule <k> require ( amount0In > 0 || amount1In > 0 , "UniswapV2: INSUFFICIENT_INPUT_AMOUNT" , .TypedVals ) ;  { uint256 balance0Adjusted = balance0 * 1000 - amount0In * 3 ;  uint256 balance1Adjusted = balance1 * 1000 - amount1In * 3 ;  require ( balance0Adjusted * balance1Adjusted >= uint256 ( reserves [ 0 ] , .TypedVals ) * reserves [ 1 ] * 1000 ** 2 , "UniswapV2: K" , .TypedVals ) ;  .Statements } fidUpdate ( balance0 , balance1 , reserves [ 0 ] , reserves [ 1 ] , .TypedVals ) ; Ss:Statements => fidUpdate ( v ( Vb0 , uint256 ) , v ( Vb1 , uint256 ) , v ( Vr0 , uint112 ) , v ( Vr1 , uint112 ) , .TypedVals ) ~> freezerExpressionStatement ( ) ~> Ss ...</k>
+       <summarize> true </summarize>
+       <env>... (amount0In |-> var ( Ia0 , uint256 ))
+                (amount1In |-> var ( Ia1 , uint256 ))
+                (balance0 |-> var ( Ib0 , uint256 ))
+                (balance1 |-> var ( Ib1 , uint256 ))
+                (reserves |-> var ( Ir , uint112 [ ] ))
+       ...</env>
+       <store> ( _ [ Ia0 <- Va0:MInt{256} ]
+                   [ Ia1 <- Va1:MInt{256} ]
+                   [ Ib0 <- Vb0:MInt{256} ]
+                   [ Ib1 <- Vb1:MInt{256} ]
+                   [ Ir <- ListItem(Vr0:MInt{112}) ListItem(Vr1:MInt{112}) ListItem(_) ]
+               ) #as STORE =>
+               STORE ListItem(Vb0 *MInt 1000p256 -MInt Va0 *MInt 3p256)
+                     ListItem(Vb1 *MInt 1000p256 -MInt Va1 *MInt 3p256)
+       </store>
+       <this-type> uniswapV2Pair </this-type>
+       <current-function> swap </current-function>
+    requires (Va0 >uMInt 0p256 orBool Va1 >uMInt 0p256)
+     andBool (((Vb0 *MInt 1000p256 -MInt Va0 *MInt 3p256) *MInt (Vb1 *MInt 1000p256 -MInt Va1 *MInt 3p256)) >=uMInt (roundMInt(Vr0)::MInt{256} *MInt roundMInt(Vr1)::MInt{256} *MInt 1000000p256)) [priority(40)]
+
+  // fidUpdate return to end of swap
+  rule <k> void ~> freezerExpressionStatement ( ) ~> emit swapEvent ( msg . sender , amount0In , amount1In , amount0Out , amount1Out , to , .TypedVals ) ;  .Statements => .K ...</k>
+       <summarize> true </summarize>
+       <msg-sender> _ </msg-sender>
```

Comment:
What we're matching here?

Reply:
After fidSwap returns, there are remaining freezers and statements until the end of function. As part of those, we are accessing certain environment variables and the sender cell. While I don't care about their values, I just matched them to make sure they are there. because they are still used to compute the arguments of the emit statement, even though we don't use them. I can skip this if you prefer.